pkg/api/converterfromapi.go

```diff
@@ -245,10 +286,30 @@ func convertPropertiesToV20170131(api *Properties, p *v20170131.Properties) {
 		convertMasterProfileToV20170131(api.MasterProfile, p.MasterProfile)
 	}
 	p.AgentPoolProfiles = []*v20170131.AgentPoolProfile{}
-	for _, apiProfile := range api.AgentPoolProfiles {
-		v20170131Profile := &v20170131.AgentPoolProfile{}
-		convertAgentPoolProfileToV20170131(apiProfile, v20170131Profile)
-		p.AgentPoolProfiles = append(p.AgentPoolProfiles, v20170131Profile)
+	// DCOS conversion logic
+	if api.OrchestratorProfile.IsDCOS() && len(api.AgentPoolProfiles) == 2 {
```

Comment:
This set of conditions is meant to be an exact inverse pair with this:

https://github.com/Azure/acs-engine/pull/787/files#diff-fab6595d1729597cff2aeb9912285552R296